src/runner/runESLint.js

```diff
@@ -96,8 +103,70 @@ const getComputedFixValue = ({ fix, quiet, fixDryRun }) => {
   return undefined;
 };
 
+const useFlatConfig = async () => {
+  if (ESLINT_SUPPORTS_FLAT_CONFIG) {
+    switch (process.env.ESLINT_USE_FLAT_CONFIG) {
+      case 'true':
+        return true;
+      case 'false':
+        return false;
+      default: {
+        const eslintFlatConfig = await findUp(
+          ESLINT_FLAT_CONFIG_DEFAULT_FILENAME,
+          { cwd: process.cwd() },
+        );
+        return !!eslintFlatConfig;
```

Comment:
Could we do `!= null` or something instead of double bang?

Reply:
Sure, done.

FWIW I wrote this function the way it is solely because it matches the [implementation](https://github.com/eslint/eslint/blob/d85efad655deacc0dc3fdbbace33307094c3b91b/lib/cli.js#L283-L301) in [eslint](https://github.com/eslint/eslint/blob/d85efad655deacc0dc3fdbbace33307094c3b91b/lib/eslint/flat-eslint.js#L266-L271) as closely as possible.  I'm happy to continue making changes here, but keep in mind that any  more changes will add cognitive overhead if they make changes there.  Also keep in mind that this is probably not the right solution anyhow, and that [hopefully eslint will expose some functionality that does this for us](https://github.com/eslint/eslint/issues/17129).

In any case, I'm happy to make any other changes you'd like, but keep in mind I didn't do it the way I did because I thought it was pretty / ideal, I did it to minimize differences with the reference implementation, and you'll need to decide if you care about that.

Reply:
fair point. thoughts on extracting all the "copied" code to its own file, including a comment pointing to where in ESLint's code it's from. Just to make future references easier

Reply:
@SimenB I think it's a great idea, however given that https://github.com/eslint/eslint/issues/17129 has traction and I have a [PR open](https://github.com/eslint/eslint/pull/17169) to implement it, at this point I figure it's probably better to just wait for that to land and swap this commit to use the soon-to-be-exposed `shouldUseFlatConfig` instead.

For backwards compatibility I'm thinking it's probably better to change https://github.com/jest-community/jest-runner-eslint/pull/176/files#diff-f5651cdb0e581e985fe919453f1a58dfba958912e3c5e7ddfb4f31057483eb17R8 to something along these lines instead:

```js
let FlatESLint;
let shouldUseFlatConfig;

try {
    eslintExperimental = require('eslint/use-at-your-own-risk');
    FlatESLint = eslintExperimental.FlatESLint;
    shouldUseFlatConfig = eslintExperimental.shouldUseFlatConfig;
} catch {
}

const eslintSupportsFlatConfig = () => FlatESLint !== undefined && shouldUseFlatConfig !== undefined;
```

That way, we wouldn't have to copy the eslint code in (I don't think we lose much by not supporting flat eslint config for intermediary versions between 8.0 and whenever https://github.com/eslint/eslint/pull/17169 gets released) and we don't have to encode version information to check for eslint compatibility, which feels cleaner to me.

Thoughts?  Cool with that plan?

Reply:
It occurs to me, though, that it's a breaking change to require eslint 8.whatever since we currently support all the current eslints - so it might be a good idea to keep this fallback implementation so that this PR can be semver-minor.

although i suppose just "not supporting flat config" in eslint 8's below 8.whatever isn't a breaking change, it's just a constraint on the new feature's usage.

Reply:
Yeah I mean, that's what I was saying in my comment, I don't think we lose all that much in not supporting the flat config (which is itself considered experimental anyways) for eslint 8.0.0 - whatever version releases my PR, and I really don't like being in the business of duplicating functionality.

The logic has also [changed at least once](https://github.com/eslint/eslint/pull/16356) through those versions (and maybe more, but I didn't look) so if we actually wanted to support all the patches/minors in between accurately, we'd need to compile those differences and account for them (although admittedly in this case I don't think it's a big deal to support the `ESLINT_USE_FLAT_COFIG` env var even though the eslint version doesn't support it, but still, I'm not sure if there were more significant changes).

I also don't think this is considered a breaking change since this functionality doesn't even currently exist in `jest-runner-eslint`, and using the implementation above it would gracefully degrade to it's exact current state on 8.x versions in between.  So it's really more of an optional new feature with a required minimum peer dependency than it is a breaking change, and there would be no reason to require a particular eslint version _unless_ someone wanted this functionality (in which case they should _already_ be using latest eslint anyways since this is pre-release experimental functionality in eslint).

Reply:
Agreed. Updating the OP in this PR to state the version requirement should be enough.

Thanks for pushing this through @cprussin!